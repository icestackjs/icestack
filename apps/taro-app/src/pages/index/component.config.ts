export default definePageConfig({
  navigationBarTitleText: '组件',
  navigationStyle: 'custom',
})
