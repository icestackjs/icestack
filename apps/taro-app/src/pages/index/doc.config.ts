export default definePageConfig({
  navigationBarTitleText: '文档',
  navigationStyle: 'custom',
})
