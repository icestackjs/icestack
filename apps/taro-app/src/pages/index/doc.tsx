import levelI18n from '#docs/_meta.zh-CN.json'

import contribute from '#docs/contribute.zh-CN.mdx'
import core from '#docs/core.zh-CN.mdx'
import level2Core18n from '#docs/core/_meta.zh-CN.json'
import composeClass from '#docs/core/composeClass.zh-CN.mdx'
import extraComponent from '#docs/core/extraComponent.zh-CN.mdx'
import override from '#docs/core/override.zh-CN.mdx'
import preset from '#docs/core/preset.zh-CN.mdx'
import theme from '#docs/core/theme.zh-CN.mdx'
import variant from '#docs/core/variant.zh-CN.mdx'
import Navbar from '@/components/Navbar'
import ThemeButton from '@/components/ThemeButton'
import ThemeProvider from '@/components/ThemeProvider'
import { Button, View } from '@tarojs/components'
import Taro, { useLoad } from '@tarojs/taro'
import { useMemo, useState } from 'react'
// https://webpack.js.org/guides/asset-modules/
import introduction from '#docs/introduction.zh-CN.mdx'
import options from '#docs/options.zh-CN.mdx'

import usage from '#docs/usage.zh-CN.mdx'
import level2Usage18n from '#docs/usage/_meta.zh-CN.json'
import cli from '#docs/usage/cli.zh-CN.mdx'
import mp from '#docs/usage/mp.zh-CN.mdx'
import nodejs from '#docs/usage/nodejs.zh-CN.mdx'
import tailwindcss from '#docs/usage/tailwindcss.zh-CN.mdx'

const map = {
  introduction,
  usage,
}
const langMap = {
  ...levelI18n,
}

for (const [k, v] of Object.entries({
  cli,
  tailwindcss,
  nodejs,
  mp,
})) {
  map[['usage', k].join('.')] = v
  langMap[['usage', k].join('.')] = level2Usage18n[k]
}
map.core = core
for (const [k, v] of Object.entries({
  variant,
  theme,
  override,
  preset,
  composeClass,
  extraComponent,
})) {
  map[['core', k].join('.')] = v
  langMap[['core', k].join('.')] = level2Core18n[k]
}
map.contribute = contribute
map.options = options

const kkkkkkks = Object.keys(map) // orderBy()

export default function Index() {
  const [id, setId] = useState('')
  const title = useMemo(() => {
    const t = langMap[id] ?? ''
    t
    && Taro.setNavigationBarTitle({
      title: t,
    })
    return t
  }, [id])

  useLoad<{ id: string }>((params) => {
    if (params.id) {
      const t = params.id
      setId(t)
      console.log(t)
    }
  })
  const content = useMemo(() => {
    return map[id] ?? ''
  }, [id])

  const { next, prev } = useMemo(() => {
    // console.log(kkkkkkks, id)
    const idx = kkkkkkks.indexOf(id)
    let prev
    let next
    if (idx > -1) {
      prev = kkkkkkks[idx - 1]
      next = kkkkkkks[idx + 1]
    }

    return {
      current: id,
      prev,
      next,
    }
  }, [id])
  // console.log(current)

  function goPrev() {
    setId(prev)
    Taro.pageScrollTo({
      scrollTop: 0,
      duration: 300,
    })
  }

  function goNext() {
    setId(next)
    Taro.pageScrollTo({
      scrollTop: 0,
      duration: 300,
    })
  }
  return (
    <ThemeProvider>
      <Navbar>
        <View className="flex h-full items-center justify-center">{title}</View>
      </Navbar>
      <View className="px-4">
        <md markdown content={content}></md>
      </View>
      <View className="flex justify-between px-4 py-2">
        <View>
          {prev && (
            <Button onClick={goPrev} className="btn text-sm">
              <View className="i-mdi-arrow-left-bold"></View>
              {langMap[prev]}
            </Button>
          )}
        </View>
        <View>
          {next && (
            <Button onClick={goNext} className="btn btn-primary text-sm">
              {langMap[next]}
              {' '}
              <View className="i-mdi-arrow-right-bold"></View>
            </Button>
          )}
        </View>
      </View>
      <ThemeButton></ThemeButton>
    </ThemeProvider>
  )
}
