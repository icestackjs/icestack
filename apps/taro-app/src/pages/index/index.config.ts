export default definePageConfig({
  navigationBarTitleText: '首页',
  navigationStyle: 'custom',
})
