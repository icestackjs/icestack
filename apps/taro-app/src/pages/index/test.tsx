import { View, Text, Button } from '@tarojs/components'
import { useLoad } from '@tarojs/taro'
import { useState } from 'react'
// import { css } from '../../../styled-system/css'
import { css } from '@styled-system/css'
import { token } from '@styled-system/tokens'

const Component = (props) => {
  return (
    <View
      className={css({
        // ✅ Good: Store the value in a CSS custom property
        color: 'var(--color)'
      })}
      style={{
        // ✅ Good: Handle the runtime value in the style attribute
        '--color': token(`colors.${props.color}`)
      }}
    >
      Dynamic color with runtime value
    </View>
  )
}

export default function Test() {
  const [runtimeColor, setRuntimeColor] = useState('pink.300')
  return (
    <View>
      <Text
        className={css({
          color: 'green.400',
          bg: 'gray.200',
          border: '1px solid token(colors.red.400,red)'
        })}
      >
        Hello world!
      </Text>
      <Component color={runtimeColor}></Component>
    </View>
  )
}
