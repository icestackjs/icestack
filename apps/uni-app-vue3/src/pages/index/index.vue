<template>
  <view class="content">
    <WeappTailwindcss></WeappTailwindcss>

    <view class="text-gray-900/50 mb-2 before:content-['当前系统主题:']">
      {{ themeRef }}
    </view>
    <view class="space-y-[20rpx] flex flex-col items-center">
      <view
        class="bg-[#010101] h-16 w-16 rounded-[20rpx] text-white flex justify-center items-center after:content-['hover']"
        hover-class="bg-[gray] after:!content-['good!']"></view>
      <view
        class="grid grid-cols-3 divide-x-[10px] divide-[#010101] divide-solid">
        <div :class="classArray">1</div>
        <div>2</div>
        <div :class="classArray">3</div>
      </view>
      <view
        class="w-32 py-2 rounded-md font-semibold text-white bg-pink-500 ring-4 ring-pink-300">
        Default
      </view>
      <view>
        <button class="text-[#fff]" :class="buttonClass" @click="increment">
          click here to inc {{ count }}
        </button>
      </view>

      <view class="test">@apply</view>
    </view>
  </view>
</template>

<script setup lang="ts">
import WeappTailwindcss from '@/components/WeappTailwindcss.vue'
import { useCounterStore } from '@/stores/counter'

const store = useCounterStore()
const { count } = storeToRefs(store)
const { increment } = store
const buttonColors = [
  'bg-[#000]',
  'bg-[#111]',
  'bg-[#222]',
  'bg-[#333]',
  'bg-[#444]',
  'bg-[#555]',
  'bg-[#666]',
  'bg-[#777]',
  'bg-[#888]',
  'bg-[#999]',
  'bg-[#aaa]',
  'bg-[#bbb]',
  'bg-[#ccc]',
  'bg-[#ddd]',
  'bg-[#eee]',
  'bg-[#fff]'
]
const title = ref('Hello')
const themeRef = ref(uni.getSystemInfoSync().theme)
const classArray = computed(() => [
  title.value ? 'bg-[#ff00ff]' : undefined,
  {
    'text-[#00ffff]': Boolean(title),
    "bg-[url('https://xxx.com/xx.webp')]": true,
    "bg-[url('https://yyyy.com/ccc.webp')]": true
  }
])
const buttonClass = computed(() => {
  return buttonColors[count.value % buttonColors.length]
})
// #ifdef MP
uni.onThemeChange(({ theme }: { theme: 'dark' | 'light' }) => {
  themeRef.value = theme
})
// #endif
onBeforeUnmount(() => {
  // #ifdef MP
  uni.offThemeChange(() => {
    console.log('offThemeChange')
  })
  // #endif
})

onLoad(() => {
  console.log('欢迎使用uni-app-vite-vue3-tailwindcss模板')
})
</script>

<style lang="scss" scoped>
.logo {
  @apply h-[100rpx] w-[100rpx];
}

.content {
  @apply flex flex-col items-center;
}

.test {
  @apply flex items-center justify-center h-[100px] w-[100px] rounded-[40px] bg-[#123456] bg-opacity-[0.54] text-[#ffffff] #{!important};
}
</style>
