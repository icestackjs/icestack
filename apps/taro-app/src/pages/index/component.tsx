import { View, Text, Button, ViewProps } from '@tarojs/components'
// import { useLoad } from '@tarojs/taro'
import React, { useState, useMemo } from 'react'
import './index.scss'
import { upperFirst } from 'lodash-es'
import ThemeProvider from '@/components/ThemeProvider'
import Taro, { useLoad } from '@tarojs/taro'

const map = {
  button: () => {
    return (
      <View>
        <View className='ice-subtitle'>按钮类型</View>
      </View>
    )
  },
  alert: () => {
    return (
      <View>
        <View className='ice-alert'>
          <Text className='i-mdi-information-outline w-6 h-6'></Text>
          <Text>默认alert</Text>
        </View>
        <View className='ice-alert ice-alert-primary'>
          <Text className='i-mdi-information-outline w-6 h-6'></Text>
          <Text>默认alert</Text>
        </View>
        <View className='ice-alert ice-alert-success'>
          <Text className='i-mdi-information-outline w-6 h-6'></Text>
          <Text>默认alert</Text>
        </View>
        <View className='ice-alert ice-alert-warning'>
          <Text className='i-mdi-information-outline w-6 h-6'></Text>
          <Text>默认alert</Text>
        </View>
      </View>
    )
  }
}

export default function Index() {
  const [mode, setMode] = useState<'light' | 'dark'>('light')
  const [com, setCom] = useState('')
  useLoad<{ id: string }>((params) => {
    if (params.id) {
      Taro.setNavigationBarTitle({
        title: upperFirst(params.id)
      })
      setCom(params.id)
    }
  })
  const TargetCom = useMemo(() => {
    return map[com]
  }, [com])
  return (
    <ThemeProvider mode={mode}>
      <View className='px-4'>{TargetCom && <TargetCom></TargetCom>}</View>
    </ThemeProvider>
  )
}
