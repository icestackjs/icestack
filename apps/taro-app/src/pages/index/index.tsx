import { View, Text } from '@tarojs/components'
import Taro from '@tarojs/taro'
import { upperFirst } from 'lodash-es'
import { ReactNode, useState } from 'react'
import './index.scss'
import ThemeProvider from '@/components/ThemeProvider'
import { group, i18n } from '../../group'
import ThemeButton from '@/components/ThemeButton'
import Navbar from '@/components/Navbar'
import Footer from '@/components/Footer'
import HomeTitle from '@/components/HomeTitle'

const aliasMap = {
  range: 'Slider',
  toggle: 'Switch'
}

function getAlias(componentName: string) {
  const x = aliasMap[componentName]
  if (x) {
    return '/' + x
  }
  return ''
}

export default function Index() {
  return (
    <ThemeProvider>
      <Navbar>
        <View
          className='text-center h-full flex pl-4 items-center text-lg'
          onClick={() => {
            Taro.setClipboardData({
              data: 'https://github.com/sonofmagic/icestack'
            })
          }}
        >
          <View className='rounded-md border border-solid border-[rgba(31,35,40,0.15)] bg-[rgb(246,248,250)] dark:border-[rgba(205,217,229,0.1)] dark:bg-[rgb(55,62,71)] flex items-center px-2 py-1'>
            <View className='i-mdi-github'></View>
            <View className='text-slate-900 dark:text-white mx-0.5 i-mdi-add'></View>
            <View className='i-mdi-star text-[#daaa3f]'></View>
          </View>
        </View>
      </Navbar>
      <View className='min-h-screen px-5 pt-2 pb-3'>
        <View className='px-4'>
          <HomeTitle />
          <View className='text-slate-600 dark:text-slate-400 text-sm mb-1 text-center'>灵活自由的开源CSS Component生成器</View>

          {/* <md markdown content={'```js\nconst y = \'1111\'\n```\n'}></md> */}
          {/* <View className='text-slate-500 dark:text-slate-300 text-xs mb-1'>tips: 本小程序由于大量使用了 grid 布局和 css 新特性，可能会出现部分机型不兼容的情况</View> */}
          {/* <View
            className='mb-4 text-xs text-slate-500 dark:text-slate-300'
            onClick={() => {
              Taro.setClipboardData({
                data: 'https://github.com/sonofmagic/icestack/tree/main/apps/taro-app'
              })
            }}
          >
            源代码地址:
            <View className='link link-primary'>sonofmagic/icestack/apps/taro-app</View>
          </View> */}
        </View>

        {Object.entries(group).reduce<ReactNode[]>((acc, [groupName, componentNames]) => {
          if (componentNames.length) {
            acc.push(<View className='text-slate-600 dark:text-slate-400 text-sm ml-4 mt-6 mb-2'>{i18n[groupName]}</View>)
            // space-y-3
            const res: ReactNode[] = []
            for (const componentName of componentNames) {
              if (['select', 'diff','tooltip'].includes(componentName)) {
                continue
              }
              res.push(
                <View
                  className='bg-gray-100 dark:bg-sky-300/[0.15] py-2.5 rounded-full pl-5 pr-3.5 flex justify-between items-center'
                  hoverClass='bg-gray-300/50 dark:bg-sky-500/50'
                  key={componentName}
                  onClick={() => {
                    Taro.navigateTo({
                      url: 'component?id=' + componentName
                    })
                  }}
                >
                  <View className='text-sm text-slate-700 dark:text-slate-400'>
                    {upperFirst(componentName)}
                    {getAlias(componentName)} {i18n[componentName]}
                  </View>
                  <View className='i-mdi-chevron-right text-slate-700 dark:text-slate-400'></View>
                </View>
              )
            }
            acc.push(<View className='space-y-3'>{res}</View>)
          }

          return acc
        }, [])}
      </View>
      <ThemeButton></ThemeButton>
      <Footer></Footer>
    </ThemeProvider>
  )
}
