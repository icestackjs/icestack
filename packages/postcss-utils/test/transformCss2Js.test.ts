import { transformCss2Js } from '@/index'

describe('transformCss2Js', () => {
  it('parses nested rules', () => {
    const root = transformCss2Js('a { b {} }')
    expect(root).toMatchSnapshot()
  })

  it('parses at-rules inside rules', () => {
    const root = transformCss2Js('a { @media {} }')
    expect(root).toMatchSnapshot()
  })

  it('parses variables', () => {
    const root = transformCss2Js('$var: 1;')
    expect(root).toMatchSnapshot()
  })

  it('parses inline comments', () => {
    const root = transformCss2Js('\n// a \n/* b */')
    expect(root).toMatchSnapshot()
  })

  it('parses empty inline comments', () => {
    const root = transformCss2Js('//\n// ')
    expect(root).toMatchSnapshot()
  })

  it('parses inline comments inside selector', () => {
    const root = transformCss2Js('a\n// c/**/\nb { }')
    expect(root).toMatchSnapshot()
  })

  it('does not parse comments inside brackets', () => {
    const root = transformCss2Js('a { cursor: url(http://ya.ru) }')
    expect(root).toMatchSnapshot()
  })

  it('parses interpolation', () => {
    const root = transformCss2Js('#{$selector}:hover { #{$prop}-size: #{$color} }')
    expect(root).toMatchSnapshot()
  })

  test('parses interpolation inside word', () => {
    const root = transformCss2Js('.#{class} {}')
    expect(root).toMatchSnapshot()
  })

  test('parses non-interpolation', () => {
    const root = transformCss2Js('\\#{ color: black }')
    expect(root).toMatchSnapshot()
  })

  test('parses interpolation inside interpolation', () => {
    const root = transformCss2Js('$column: #{"#{&}__column"};')
    expect(root).toMatchSnapshot()
  })

  test("parses interpolation that's the entire at-rule", () => {
    const root = transformCss2Js('@#{$var} param { }')
    expect(root).toMatchSnapshot()
  })

  test('parses interpolation at the beginning of at-rule', () => {
    const root = transformCss2Js('@#{$var}suffix param { }')
    expect(root).toMatchSnapshot()
  })

  test('parses interpolation within at-rule', () => {
    const root = transformCss2Js('@before#{$var}after param { }')
    expect(root).toMatchSnapshot()
  })

  test('parses interpolation right after at-rule', () => {
    const root = transformCss2Js('@media#{$var} { }')
    expect(root).toMatchSnapshot()
  })

  test('parses interpolation in at-rule value', () => {
    const root = transformCss2Js('@media #{$var} { }')
    expect(root).toMatchSnapshot()
  })

  test('parses interpolation in url()', () => {
    const root = transformCss2Js('image: url(#{get(path)}.png)')
    expect(root).toMatchSnapshot()
  })

  test('parses text in rules', () => {
    const root = transformCss2Js('a { margin:text { left: 10px; }}')
    expect(root).toMatchSnapshot()
  })

  test('parses semicolon in rules', () => {
    const root = transformCss2Js('a { test(a: 1) { left: 10px; }}')
    expect(root).toMatchSnapshot()
  })

  test('parsers prefixed pseudo in rules', () => {
    const root = transformCss2Js('input:-moz-focusring { left: 1px }')
    expect(root).toMatchSnapshot()
  })

  // test('parses nested props as rule', () => {
  //   const root = transformCss2Js('a { margin: { left: 10px; }}')
  //   expect(root).toMatchSnapshot()
  // })

  test('parses nested props with value', () => {
    const root = transformCss2Js('a { margin: 0 { left: 10px; }}')
    expect(root).toMatchSnapshot()
  })

  test('parses nested props with space-less digit', () => {
    const root = transformCss2Js('a { margin:0 { left: 10px; }}')
    expect(root).toMatchSnapshot()
  })

  test('parses nested props with new line as rule', () => {
    const root = transformCss2Js('a { \n margin  \n:0 { left: 10px; }}')
    expect(root).toMatchSnapshot()
  })

  test('parses nested props with important', () => {
    const root = transformCss2Js('a { margin: 0!important { left: 10px; }}')
    expect(root).toMatchSnapshot()
  })

  test('parses interpolation with variable', () => {
    const root = transformCss2Js('&:#{$var} {}')
    expect(root).toMatchSnapshot()
  })

  test('parses comment inside comment', () => {
    const root = transformCss2Js('a {\n//a/*b*/c\n}')
    expect(root).toMatchSnapshot()
  })

  test('parses complex interpolation', () => {
    const root = transformCss2Js('content: #{fn("\\"}")};')
    expect(root).toMatchSnapshot()
  })

  test('parses interpolation inside string', () => {
    const root = transformCss2Js('content: "#{fn("\\"}")}";')
    expect(root).toMatchSnapshot()
  })
})
