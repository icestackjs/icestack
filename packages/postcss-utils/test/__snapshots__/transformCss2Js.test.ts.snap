// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`transformCss2Js > does not parse comments inside brackets 1`] = `
{
  "a": {
    "css": {
      "cursor": "url(http://ya.ru)",
    },
  },
}
`;

exports[`transformCss2Js > parsers prefixed pseudo in rules 1`] = `
{
  "input:-moz-focusring": {
    "css": {
      "left": "1px",
    },
  },
}
`;

exports[`transformCss2Js > parses at-rules inside rules 1`] = `
{
  "a": {
    "@media ": {},
  },
}
`;

exports[`transformCss2Js > parses comment inside comment 1`] = `
{
  "a": {},
}
`;

exports[`transformCss2Js > parses complex interpolation 1`] = `
{
  "css": {
    "content": "#{fn("\\"}")}",
  },
}
`;

exports[`transformCss2Js > parses empty inline comments 1`] = `{}`;

exports[`transformCss2Js > parses inline comments 1`] = `{}`;

exports[`transformCss2Js > parses inline comments inside selector 1`] = `
{
  "a b": {},
}
`;

exports[`transformCss2Js > parses interpolation 1`] = `
{
  "#{$selector}:hover": {
    "css": {
      "#{$prop}-size": "#{$color}",
    },
  },
}
`;

exports[`transformCss2Js > parses interpolation at the beginning of at-rule 1`] = `
{
  "@#{$var}suffix param": {},
}
`;

exports[`transformCss2Js > parses interpolation in at-rule value 1`] = `
{
  "@media #{$var}": {},
}
`;

exports[`transformCss2Js > parses interpolation in url() 1`] = `
{
  "css": {
    "image": "url(#{get(path)}.png)",
  },
}
`;

exports[`transformCss2Js > parses interpolation inside interpolation 1`] = `
{
  "css": {
    "$column": "#{"#{&}__column"}",
  },
}
`;

exports[`transformCss2Js > parses interpolation inside string 1`] = `
{
  "css": {
    "content": ""#{fn("\\"}")}"",
  },
}
`;

exports[`transformCss2Js > parses interpolation inside word 1`] = `
{
  ".#{class}": {},
}
`;

exports[`transformCss2Js > parses interpolation right after at-rule 1`] = `
{
  "@media#{$var} ": {},
}
`;

exports[`transformCss2Js > parses interpolation that's the entire at-rule 1`] = `
{
  "@#{$var} param": {},
}
`;

exports[`transformCss2Js > parses interpolation with variable 1`] = `
{
  "&:#{$var}": {},
}
`;

exports[`transformCss2Js > parses interpolation within at-rule 1`] = `
{
  "@before#{$var}after param": {},
}
`;

exports[`transformCss2Js > parses nested props with important 1`] = `
{
  "a": {
    "css": {
      "margin": "0 !important",
    },
  },
}
`;

exports[`transformCss2Js > parses nested props with new line as rule 1`] = `
{
  "a": {
    "margin :0": {
      "css": {
        "left": "10px",
      },
    },
  },
}
`;

exports[`transformCss2Js > parses nested props with space-less digit 1`] = `
{
  "a": {
    "css": {
      "margin": "0",
    },
  },
}
`;

exports[`transformCss2Js > parses nested props with value 1`] = `
{
  "a": {
    "css": {
      "margin": "0",
    },
  },
}
`;

exports[`transformCss2Js > parses nested rules 1`] = `
{
  "a": {
    "b": {},
  },
}
`;

exports[`transformCss2Js > parses non-interpolation 1`] = `
{
  "\\#": {
    "css": {
      "color": "black",
    },
  },
}
`;

exports[`transformCss2Js > parses semicolon in rules 1`] = `
{
  "a": {
    "test(a: 1)": {
      "css": {
        "left": "10px",
      },
    },
  },
}
`;

exports[`transformCss2Js > parses text in rules 1`] = `
{
  "a": {
    "margin:text": {
      "css": {
        "left": "10px",
      },
    },
  },
}
`;

exports[`transformCss2Js > parses variables 1`] = `
{
  "css": {
    "$var": "1",
  },
}
`;
